src/DynamoAnalyzer/Helper/GregHelper.cs

```diff
@@ -0,0 +1,241 @@
+using DynamoAnalyzer.Models;
+using Greg;
+using Greg.Requests;
+using Greg.Responses;
+
+namespace DynamoAnalyzer.Helper
+{
+    /// <summary>
+    /// Provides methos to manage packages
+    /// </summary>
+    public static class GregHelper
+    {
+        private static GregClient _client;
+
+        public static GregClient GetClient()
+        {
+            return _client ??= new GregClient(new Dynamo.Core.IDSDKManager(), ConfigHelper.GetConfiguration()["DynamoPackagesURL"]);
+        }
+
+        /// <summary>
+        /// Downloads the full package list
+        /// </summary>
+        /// <returns></returns>
+        public static async Task<List<PackageHeader>> GetPackages()
+        {
+            ResponseWithContentBody<List<PackageHeader>> pkgResponse = null;
+
+            LogHelper.Log("PackageList", "Download start");
+            await Task.Run(() =>
+            {
+                pkgResponse = GetClient().ExecuteAndDeserializeWithContent<List<PackageHeader>>(HeaderCollectionDownload.ByEngine(ConfigHelper.GetConfiguration()["Engine"]));
+            });
+
+            CleanPackagesWithWrongVersions(pkgResponse.content);
+
+            List<PackageHeader> response = pkgResponse.content.Where(f => !f.banned).ToList();
```

Comment:
are you actually getting banned packages in the response?

Reply:
no banned packages at this moment, but 542 deprecated packages, added an additional check to ignore deprecated packages